packages/plugin-directives/src/global-types.ts

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import {
+  EnumValues,
+  FieldNullability,
+  FieldRequiredness,
+  InputFieldMap,
+  InputType,
+  InterfaceParam,
+  ObjectParam,
+  RootName,
+  SchemaTypes,
+  TypeParam,
+} from '@giraphql/core';
+import { SchemaDirectiveVisitor } from '@graphql-tools/utils';
+import { GraphQLSchema } from 'graphql';
+import DirectivePlugin from '.';
+import { DirectiveLocation, Directives } from './types';
+
+declare global {
+  export namespace GiraphQLSchemaTypes {
+    export interface Plugins<Types extends SchemaTypes> {
+      GiraphQLDirectives: DirectivePlugin<Types>;
+    }
+
+    export interface TypeInfo {
+      Directives: Record<
+        string,
+        {
+          locations: DirectiveLocation;
+          args?: Record<string, unknown>;
+        }
+      >;
+    }
+
+    export interface ExtendDefaultTypes<PartialTypes extends Partial<TypeInfo>> {
+      Directives: undefined extends PartialTypes['Directives']
+        ? {}
+        : PartialTypes['Directives'] & {};
+    }
+
+    export interface SchemaBuilderOptions<Types extends SchemaTypes> {
+      schemaDirectives?: Record<keyof Types['Directives'], typeof SchemaDirectiveVisitor>;
+      schemaTransforms?: ((schema: GraphQLSchema) => GraphQLSchema)[];
```

Comment:
these can't be applied via a plugin, directives that use schemaTransform will need to manual invoke the transforms on the schema after it is returned.  (remove and document)